frontend/app/routes/protected/person-case/name-info.tsx

```diff
@@ -0,0 +1,226 @@
+import { useState } from 'react';
+
+import { data, useFetcher } from 'react-router';
+import type { RouteHandle } from 'react-router';
+
+import { faExclamationCircle, faXmark } from '@fortawesome/free-solid-svg-icons';
+import type { SessionData } from 'express-session';
+import { useTranslation } from 'react-i18next';
+import * as v from 'valibot';
+
+import type { Route, Info } from './+types/name-info';
+
+import { requireAuth } from '~/.server/utils/auth-utils';
+import { i18nRedirect } from '~/.server/utils/route-utils';
+import { Button } from '~/components/button';
+import { ErrorSummary } from '~/components/error-summary';
+import { InputCheckboxes } from '~/components/input-checkboxes';
+import { InputField } from '~/components/input-field';
+import { InputRadios } from '~/components/input-radios';
+import { PageTitle } from '~/components/page-title';
+import { Progress } from '~/components/progress';
+import { AppError } from '~/errors/app-error';
+import { getFixedT } from '~/i18n-config.server';
+import { handle as parentHandle } from '~/routes/protected/layout';
+import { getLanguage } from '~/utils/i18n-utils';
+import { REGEX_PATTERNS } from '~/utils/regex-utils';
+
+type NameInfoSessionData = NonNullable<SessionData['inPersonSINCase']['nameInfo']>;
+
+const VALID_DOC_TYPES: string[] = [
+  'marriage-document',
+  'divorce-decree',
+  'name-change',
+  'adoption-order',
+  'notarial-certificate',
+  'resident-record',
+  'replace-imm1442',
+  'birth-certificate',
+  'citizenship-certificate',
+] as const;
+
+export const handle = {
+  i18nNamespace: [...parentHandle.i18nNamespace, 'protected'],
+} as const satisfies RouteHandle;
+
+export async function loader({ context, request }: Route.LoaderArgs) {
+  requireAuth(context.session, new URL(request.url), ['user']);
+  const t = await getFixedT(request, handle.i18nNamespace);
+
+  return {
+    documentTitle: t('protected:primary-identity-document.page-title'),
+    defaultFormValues: {
+      ...context.session.inPersonSINCase?.nameInfo,
+    },
+  };
+}
+
+export function meta({ data }: Route.MetaArgs) {
+  return [{ title: data.documentTitle }];
+}
+
+export async function action({ context, request }: Route.ActionArgs) {
+  requireAuth(context.session, new URL(request.url), ['user']);
+  const lang = getLanguage(request);
+  const t = await getFixedT(request, handle.i18nNamespace);
+
+  const formData = await request.formData();
+  const action = formData.get('action');
+  const docsRequired = formData.get('docs-required') === 'Yes';
+  const maxLength = 100;
```

Comment:
This could have a better name. What's it max length of? I had to scroll down to see how it was used to know.